docs/Technical-Documentation/tech-memos/priotitized-errors/prioritized-errors.md

```diff
@@ -0,0 +1,101 @@
+# TDP Prioritized Parser Errors
+
+**Audience**: TDP Software Engineers <br>
+**Subject**:  Prioritized Errors <br>
+**Date**:     October 20, 2024 <br>
+
+## Summary
+This technical memorandum provides a suggested path to implement a set of new requirements OFA has generated to alleviate the sheer number of parser errors generated during a STT's data submission. OFA has indicated, while still necessary for admins to see, some errors are of a lower priority for STTs to view and even prevent them from using the submission history reports to correct their submissions. Thus, the OFA team has requested that a "priority" be assigned to parser errors so that the report STTs receive is filtered down to only the highest priority errors that must be fixed. Regardless of how errors are prioritized, OFA admins will still retain the ability to generate the unfiltered error report for further, more in-depth investigation if a necessary situation presents itself.
+
+## Background
+Currently, error reports are generated in the TDP backend via the `get_xls_serialized_file` function. This function accepts the serialized queryset of the appropriate `ParserError`s queryset. This function the writes an XLSX file and returns it to the user. Apart from the lack of priotization in the report generated from this function, it also introduces the possibility to cause an out of memory (OOM) error. This can occur because, the Django model serializer brings the entire queryset into memory to serialize it into JSON. Because these ParserError querysets can be very large (hundreds of thousands), we will also alleviate the memory pressure `get_xls_serialized_file` introduces by removing the Django model serializer and make use of queryset pagination.
+
+## Out of Scope
+Current requirements from OFA do not require category two errors to be queryable by value and expected value. That feature is out of scope within the tech memo and would require more design and implementation work.
```

Comment:
@elipe17 is this true? can you remind me what this is in reference to?

Reply:
For the time being it is true. the OneNote did not specify specific values of the cat2 validators were of interest but that all values of the cat2 validator were of interest. I briefly asked you about it in our MM chat attached below.
![Screenshot 2024-12-05 at 10 12 13 AM](https://github.com/user-attachments/assets/fb4f34a8-d734-4c98-988a-0c8ce23a69ef)

Reply:
Just for your SA, those messages are from October 18